privval/file_test.go

```diff
@@ -120,34 +125,34 @@ func TestUnmarshalValidatorKey(t *testing.T) {
 	privKey := ed25519.GenPrivKey()
 	pubKey := privKey.PubKey()
 	addr := pubKey.Address()
-	pubBytes := []byte(pubKey.(ed25519.PubKey))
-	privBytes := []byte(privKey)
+	pubBytes := pubKey.Bytes()
+	privBytes := privKey.Bytes()
 	pubB64 := base64.StdEncoding.EncodeToString(pubBytes)
 	privB64 := base64.StdEncoding.EncodeToString(privBytes)
 
 	serialized := fmt.Sprintf(`{
   "address": "%s",
-  "pub_key": {
-    "type": "tendermint/PubKeyEd25519",
-    "value": "%s"
-  },
-  "priv_key": {
-    "type": "tendermint/PrivKeyEd25519",
-    "value": "%s"
-  }
+  "pubKey": {"ed25519":"%s"},
```

Comment:
what is the preferred way of handling this transition for validators? should we maintain backwards compatibility or have a migration path.

cc @melekes @erikgrinaker @tessr

Reply:
I'm curious about this too - it has implications for the RPC API as well, and a few other places. If we're going to break this, we might as well do it now since we're breaking everything else with Protobuf, but it's a lot of breakage to deal with.

Reply:
Having given this some thought, I think the new scheme makes more sense (using a key per variant instead of a `type` field), since it can represent an actual well-defined schema instead of having a random assortment of fields depending on the `type` field. This also simplifies decoding, since we don't have to do it in two passes (first to get the `type` field, then to decode the contents). My vote is 👍.

Reply:
should be fine as long as there's an upgrade path